lib/googlestaticmap.rb

```diff
@@ -117,7 +117,7 @@ def url(protocol='http')
     attrs << ["key", @api_key] if !@api_key.nil?
     attrs << ["client", @client_id] if @api_key.nil? && !@client_id.nil? && !@private_key.nil?
     path << attrs.sort_by {|k,v| k}.collect {|attr| "#{attr[0]}=#{attr[1]}"}.join("&")
-    if @api_key.nil? && !@client_id.nil? && !@private_key.nil?
+    if (!@api_key.nil? ^ !@client_id.nil?) && !@private_key.nil?
```

Comment:
The reason was the one you just mentioned, its not a real use case and i thought it could help the developer to detect a possible error, but i can change it if you think its unnecessary type of validation.